src/usecases/__tests__/courseListToSpecialTable.test.ts

```diff
@@ -0,0 +1,559 @@
+  room: "8B204",
+  id: "xxx",
+};
+const res2: { module: string[]; name: string; room: string; id: string } = {
+  module: ["春BC", "夏休"],
+  name: "日本美術特別技法演習-1",
+  room: "アート&デザイン実習室",
+  id: "xxx",
+};
+const res3: { module: string[]; name: string; room: string; id: string } = {
+  module: ["秋ABC"],
+  name: "原子核実験物理学II",
+  room: "自然B118",
+  id: "xxx",
+};
+const res4: { module: string[]; name: string; room: string; id: string } = {
+  module: ["通年"],
+  name: "国際生命産業科学インターンシップ",
+  room: "",
+  id: "xxx",
+};
+const res5: { module: string[]; name: string; room: string; id: string } = {
+  module: ["通年"],
+  name: "応用動物昆虫学特別研究I",
+  room: "研究室",
+  id: "xxx",
+};
+const res6: { module: string[]; name: string; room: string; id: string } = {
+  module: ["春C", "夏休"],
+  name: "色彩学",
+  room: "3A棟, 中央図書館",
+  id: "xxx",
+};
+const res7: { module: string[]; name: string; room: string; id: string } = {
+  module: ["通年"],
+  name: "International Conference Seminar I",
+  room: "",
+  id: "xxx",
+};
+const res8: { module: string[]; name: string; room: string; id: string } = {
+  module: ["春ABC"],
+  name: "International Conference Seminar I",
+  room: "情報理工後期特別研究室",
+  id: "xxx",
+};
+const res9: { module: string[]; name: string; room: string; id: string } = {
+  module: ["夏休", "秋C", "春休"],
+  name: "生物機械工学",
+  room: "2D306",
+  id: "xxx",
+};
+const res10: { module: string[]; name: string; room: string; id: string } = {
+  module: ["通年"],
+  name: "教職論I",
+  room: "1A101",
+  id: "xxx",
+};
+const res11: { module: string[]; name: string; room: string; id: string } = {
+  module: ["夏休"],
+  name: "住環境計画概論",
+  room: "3A403",
+  id: "xxx",
+};
+const res12: { module: string[]; name: string; room: string; id: string } = {
+  module: ["通年"],
+  name: "スポーツ倫理学",
+  room: "5C416",
+  id: "xxx",
+};
+const res13: { module: string[]; name: string; room: string; id: string } = {
+  module: ["通年"],
+  name: "アジア文化資料研究演習III",
+  room: "2D305",
+  id: "xxx",
+};
+const res14: { module: string[]; name: string; room: string; id: string } = {
+  module: ["通年"],
+  name: "English Reading Skills I",
+  room: "共同利用棟A102",
+  id: "xxx",
+};
+const res15: { module: string[]; name: string; room: string; id: string } = {
+  module: ["春ABC", "秋ABC"],
+  name: "発展体育接待ゴルフ",
+  room: "人社A201, 1A101, 第1多目的道場, 3A403, 共同利用棟A102",
+  id: "xxx",
+};
+
+describe(courseListToSpecialTable.name, () => {
+  it("create spcial table.", () => {
+    const courses = [
+      course1,
+      course2,
+      course3,
+      course4,
+      course5,
+      course6,
+      course7,
+      course8,
+      course9,
+      course10,
+      course11,
+      course12,
+      course13,
+      course14,
+      course15,
+    ];
+    const table: SpecialTable = {
+      Intensive: [res15, res2, res11, res1, res3],
+      Appointment: [res14, res13, res12, res4, res5, res10],
+      AnyTime: [res7, res8, res6, res9],
+    };
+    expect(courseListToSpecialTable(courses)).toEqual(table);
```

Comment:
今回は全てのプロパティで深いマッチをさせたいので、`toEqual`にしたいです。
### toEqual
Use .toEqual to compare recursively all properties of object instances (also known as "deep" equality).
### toMatchObject
Use .toMatchObject to check that a JavaScript object matches a subset of the properties of an object.
It will match received objects with properties that are not in the expected object.

Reply:
了解！